### ISSUE TITLE
[BUG] Compile error when building traefik with go 1.22.7

### ISSUE THREAD

**Issue by eliottness** [DataDog/orchestrion#298]
**Version of orchestrion**
`v0.9.0`

**Describe what happened:**

I was reviewing #292 when I saw this after a fresh clone of `https://github.com/traefik/traefik.git` on commit `d317cd90fcb579722ab29c35410c10b6a135f949` (main)

![image](https://github.com/user-attachments/assets/01d7ea1f-2e12-4991-80ba-55515330eabd)

**Describe what you expected:**

I was expection the build to succeed because `go build .` does

**Steps to reproduce the issue:**

**Additional environment details (Version of Go, Operating System, etc.):**

**Comment by RomainMuller** [comment]
Yeah so `traefik`'s `go.mod` has `go 1.23` so it's expected it won't build wth `go 1.22`... In this case, the `go` toolchain automatically uses a more recent compiler to "make it work" (because `GOTOOLCHAIN` was not set to `local`?).

The issue here is that while we end up using a `go 1.23` compiler, the go AST & type checker package being used by `orchestrion` is the one it was built with, in this case `1.22`, which prevents it from properly loading up in this case.

Short term, we should detect this situation and emit a clear warning; or bail out with a clear error indicating what the user should do (likely, re-build their local orchestrion with the proper toolchain).

Longer term, we might consider re-spawning `orchestrion` with the correct toolchain, since we have a full compiler toolchain available anyway...

**Comment by eliottness** [comment]
Short term answer is to upgrade you Go toolchain and to reinstall orchestrion with it.
